src/js/tabs/osd.js

```diff
@@ -686,8 +687,7 @@ OSD.loadDisplayFields = function() {
             draw_order: 810,
             positionable: true,
             preview: function (osd_data) {
-                return FONT.symbol(SYM.SPEED) + ' 40' + (osd_data.unit_mode === 0 ? FONT.symbol(SYM.MPH) : FONT.symbol(SYM.KPH));
-            }
+                return FONT.symbol(SYM.SPEED) + '40' + (osd_data.unit_mode === 0 || osd_data.unit_mode === 1 ? FONT.symbol(SYM.MPH) : FONT.symbol(SYM.KPH));            }
```

Comment:
This is going bigger.... maybe is better to use several lines, one to select the unit and another one to return the string.